docs/workforce-management-framework/topics/con-Security-Strategy-for-PassportAuth.adoc

```diff
@@ -0,0 +1,52 @@
+[id='Security-Strategy-Passport.js-{chapter}']
+= Security Strategy for {WFM-RC-NameShort} PassportAuth
+
+== Session Management
+Users sessions are stored using `express-session` module.
+Visit link:https://github.com/expressjs/session[express-session] to find more information about the available express
+session options.
+
+== Access Control
+The PassportAuth module allows role based access control. Routes are protected by role as shown in the example.
+See authService is an implementation of the
+link:++../../../api/{WFM-RC-Api-Version}/auth-passport/docs/classes/_src_auth_passportauth_.passportauth.html++[passportAuth]
+in the reference application.
+
+[source,typescript]
+----
+app.get('/testAdminEndpoint', authService.protect('admin'), (req: express.Request, res: express.Response) => {
+  res.json({ routeName: '/testAdminEndpoint', msg: 'authorized to access admin endpoint' });
+});
+
+----
+Demo roles are defined as an array in link:https://github.com/feedhenry-raincatcher/raincatcher-core/blob/master/demo/data/src/users.json[user.json].
+
+
+== Password Storage
+
+Passport module has build UserRepository interface that can be used to point to user data.
+Passwords are stored in datasource that is being configured by developers (possibly external service).
+To point Passport to user datasource you need to implement the
+link:++../../../api/{WFM-RC-Api-Version}/auth-passport/docs/interfaces/_src_user_userrepository_.userrepository.html#getuserbylogin++[user repository interface]
+
+== Authentication
+From a development point of view {WFM-RC-NameShort} uses the Default Strategy for username/password authentication. Authentication is done on the cloud
+server. Passport authentication service is using a default custom strategy that is based on Passport.js local strategy.
+Passport's local strategy is agnostic about database type.
+
+Passports using
+link:++../../../api/{WFM-RC-Api-Version}/auth-passport/docs/interfaces/_src_auth_passportauth_.endpointsecurity.html#authenticate++[authenticate] middleware for accepting user credentials and creates session entry that holds user.
+To access routes user need to login using their credentials. Cookie containing user session key will be used to
+authenticate and authorize user requesting server endpoint.
+
+NOTE: For demo the login post route has an authenticate method that checks the userId and password
+against the stored user.json file it also checks the user role against the stored role on user.json.
+
+See link:{WFM-RC-Github-Core}{WFM-RC-Branch}{WFM-RC-PassportAuth-Example}[example]
+
+== Authorization
+Authorization is being supported thanks to
+link:++../../../api/{WFM-RC-Api-Version}/auth-passport/docs/interfaces/_src_auth_passportauth_.endpointsecurity.html#protect++[protect]
+middleware that validates user session id against session and checks permissions.
+
+
```

Comment:
The Session Management, Password Storage, Authentication and Authorization have already been explained elsewhere, so I don't see the added value for them in this section. The access control section is fine because it provides an example of a concept (authorisation) that was defined in another section before, but the other sections are repeating what already has been said or what is available in the linked resources that were stated already.

Reply:
This file was first and is the meat of the ticket

Reply:
I think the paths are a bit crossed here. The Security Strategy jiras are different from the Security Developer Docs - there should be other tickets for these. We should focus on usage of the two auth modules here.

The strategy guide is something that I was going to work on after performing a pentest, the strategy guide may or may not be in the open too. There is a separate PRD item (PRD-115) for this strategy guide alone.